### PR TITLE
feat(provider): add profile name from configuration file support

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## Why?
With the arrival of the new project functionality on the platform, I ended up needing to use two authenticated `scaleway` providers on two different projects within the same Terraform project.

My intention here is to provide a way to easily manage multiple profiles declared in the Scaleway CLI static configuration file in the same way as with [the AWS provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#shared-credentials-file) 

### Real use case example

```hcl
terraform {
  required_providers {
    scaleway = {
      source  = "local/scaleway/scaleway"
      version = "~> 2.1.0"
    }
  }

  required_version = ">= 0.13"
}

provider "scaleway" {
  profile = "myProjectProfile"
}

provider "scaleway" {
  alias   = "dns"
  profile = "myDnsProfile"
}

resource "scaleway_domain_record" "project_domain_record" {
  provider = scaleway.dns
  name     = "scaleway_test"
  type     = "A"
  data     = scaleway_instance_ip.server_ip.address
  dns_zone = "projects.mydomain.io"
}

resource "scaleway_instance_ip" "server_ip" {}

# ... and my other project dedicated resources...
```

> Don't hesitate to tell me if I forgot to update any other file, I would love to be able to use this feature from the official version of the provider. Hopefully this will help other users of the platform too 🤗

### CODE REVIEW

**Comment by alexandrebouthinon** [comment]
It seems [previous workflow error](https://github.com/scaleway/terraform-provider-scaleway/actions/runs/1276136409) is not related to my changes. I just amend my last commit to trigger another

**Comment by Monitob** [comment]
LGTM. You should update the sdk with this [commit](https://github.com/scaleway/scaleway-sdk-go/commit/6ee2d83584a9a661f2c7c28150d6f003d5e98ffc) 

Thank your for  your contribution.

File: docs/index.md [hunk 1]
```diff
@@ -178,6 +178,33 @@ If it fails to detect credentials inline, or in the environment, Terraform will
...
+If your shared configuration file contains:
+
+```yaml
+profiles:
+  myProfile:
+    access_key: xxxxxxxxxxxxxxxxxxxx
+    secret_key: xxxxxxxx-xxx-xxxx-xxxx-xxxxxxxxxxx
+    default_organization_id: xxxxxxxx-xxx-xxxx-xxxx-xxxxxxxxxxx 
+    default_project_id: xxxxxxxx-xxx-xxxx-xxxx-xxxxxxxxxxx
+    default_zone: fr-par-1
```
Comment (Monitob) [inline docs/index.md:194]:
```suggestion
    default_zone: fr-par-2
```

File: docs/index.md [hunk 2]
```diff
@@ -178,6 +178,33 @@ If it fails to detect credentials inline, or in the environment, Terraform will
...
+    api_url: https://api.scaleway.com
+    insecure: false
+```
+
+You can invoke and use this profile in the provider declaration:
+
+```hcl
+provider "scaleway" {
+  profile = "myProfile"
+}
```
Comment (Monitob) [inline docs/index.md:205]:
```suggestion
provider "scaleway" {
  alias   = "p2"
  profile = "myProfile"
}

resource "scaleway_instance_ip" "server_ip" {
  provider = scaleway.p2
}
```

**Review by Monitob — DISMISSED** [review]

**Review by Monitob — APPROVED** [review]

**Pull request merged**